tesseract_environment/src/ofkt/ofkt_state_solver.cpp

```diff
@@ -556,6 +548,111 @@ void OFKTStateSolver::update(EnvState& state,
     update(state, child, parent_world_tf, update_required);
 }
 
+void OFKTStateSolver::moveLinkHelper(std::vector<tesseract_scene_graph::Joint::ConstPtr>& new_kinematic_joints,
+                                     std::vector<std::string>& removed_joints,
+                                     std::vector<long>& removed_joints_indices,
+                                     const tesseract_scene_graph::Joint::ConstPtr& joint)
+{
+  auto* old_node = link_map_[joint->child_link_name];
+  const std::string& old_joint_name = old_node->getJointName();
+  old_node->getParent()->removeChild(old_node);
+
+  // Store the remove joints to process later
+  auto it = std::find(joint_names_.begin(), joint_names_.end(), old_joint_name);
+  if (it != joint_names_.end())
+  {
+    removed_joints.push_back(old_joint_name);
+    removed_joints_indices.push_back(std::distance(joint_names_.begin(), it));
+  }
+
+  current_state_->joints.erase(old_joint_name);
+  current_state_->joint_transforms.erase(old_joint_name);
+
+  nodes_.erase(old_joint_name);
+
+  addNode(joint, joint->getName(), joint->parent_link_name, joint->child_link_name, new_kinematic_joints);
+}
+
+void OFKTStateSolver::replaceJointHelper(std::vector<tesseract_scene_graph::Joint::ConstPtr>& new_kinematic_joints,
+                                         const tesseract_scene_graph::Joint::ConstPtr& joint)
+{
+  auto& n = nodes_[joint->getName()];
+
+  if (n->getType() == joint->type)
+  {
+    n->getParent()->removeChild(n.get());
+    n->setStaticTransformation(joint->parent_to_joint_origin_transform);
+    OFKTNode* new_parent = link_map_[joint->parent_link_name];
+    n->setParent(new_parent);
+    new_parent->addChild(n.get());
```

Comment:
So the state solver allows the child to be different?

Reply:
This is being checked prior to reaching this function but it may be good to add asserts here because it should never happen if the upstream checking is correct.